### ISSUE TITLE
When left running 24x7, sometimes A JavaScript error is found on Monday and requires clearing to print

### ISSUE THREAD

**Issue by evilsizorm** [ExLibrisGroup/alma-print-daemon#19]
This morning a dialog box showed indicating, "A JavaScript error occurred in the main process".  I have attached an image.  Even thought it looks like the daemon is running.  Our staff reported that printing did not resume until the error was cleared.

TypeError: Cannot read property '0' of undefined at Function.<anonymous>

Mark
![AlmaPrintDaemonJavaScriptError](https://user-images.githubusercontent.com/5598657/101361002-03688a80-3864-11eb-9913-e0592d59b9e0.png)

**Comment by mgobat** [comment]
Hmmm....I wonder if this was due to the new Alma release over the weekend....and there's a place in the APD code  that still poses a problem.

**Comment by evilsizorm** [comment]
This morning we experienced a similar occurrence. Perhaps there was server side maintenance this weekend as well.

Mark
![Alma Printer Daemon Error 2021 01 11](https://user-images.githubusercontent.com/5598657/104218609-987e1680-5402-11eb-8a82-d5bd8ae151d4.jpg)

**Comment by mgobat** [comment]
Yes, there was a new release over the weekend.  I'll look into it again.  Thanks.

**Comment by mgobat** [comment]
Mark...is there anything in the log file from around the time this last happened?  The file would be located in the username/AppData/Roaming/alma-print-daemon directory.  Considering the screenshot you included, I'm going to guess no. 
Thanks.  -m

**Comment by evilsizorm** [comment]
This morning when we arrived the Printer Daemon was in an error state with different error messages than previously.  See attached.

![Print Daemon Error 2021 03 08](https://user-images.githubusercontent.com/5598657/110333702-87f3b200-7fe7-11eb-8284-93640f265dc2.jpg)

**Comment by evilsizorm** [comment]
This morning a javascript error was showing in a dialog box.  Once we okayed it, printing resumed.
![Alma Print Daemon 2021 04 05](https://user-images.githubusercontent.com/5598657/113577653-34c14f00-95e7-11eb-93e5-73a9046c8026.jpg)

**Comment by mgobat** [comment]
That's progress...we got a line number in the error this time.  :)
Any errors logged in your log.alma-print-daemon.* this time?

**Comment by evilsizorm** [comment]
Mark there was an error message:

Error on 2021-04-04T02:17:52.204Z':  503 Service Unavailable - General error with the API Gateway – please retry later.

**Comment by evilsizorm** [comment]
After the weekend update we received a similar error this morning.
![Alma Printer Daemon 2021 05 03](https://user-images.githubusercontent.com/5598657/116941640-b41d5f00-ac35-11eb-8d12-8096a1881be8.png)

**Comment by markcorry** [comment]
We are seeing this error now, too! 
![PXL_20210902_190611786](https://user-images.githubusercontent.com/48454316/131907909-66fa07e3-a51a-4eae-92b5-ddb2008bef4b.jpg)

**Comment by mgobat** [comment]
@markcorry, with which version of the Alma Print Daemon? If pre-2.0, please try the 2.0 beta. Thanks.

**Comment by daniels** [comment]
> with which version of the Alma Print Daemon? If pre-2.0, please try the 2.0 beta. Thanks.

We just implemented Alma Print Daemon and got this error. However my IT department was reluctant to install a beta version, so we settled on 1.3.1, the latest stable release. Are bugs only fixed on the beta? If so, what makes a release stable? Is there a plan for a new stable release?

**Comment by mgobat** [comment]
@daniels, version 1.3.1 is no longer receiving updates. 2.0 added the ability the install the Alma Print Daemon as a Windows service, Any new features and bug fixes will happen in 2.0. Its been in use for quite a while by some Alma customers. We were hoping for additional feedback on the Windows Service feature before it comes out of beta.

**Comment by mgobat** [comment]
@evilsizorm, please give the latest beta a try, if you like: https://github.com/ExLibrisGroup/alma-print-daemon/releases/tag/v2.2.0-beta-02
